mlir/include/mlir/Dialect/SPIRV/IR/SPIRVBase.td

```diff
@@ -4058,7 +4058,12 @@ def SPIRV_Int32 : TypeAlias<I32, "Int32">;
 def SPIRV_Float32 : TypeAlias<F32, "Float32">;
 def SPIRV_Float : FloatOfWidths<[16, 32, 64]>;
 def SPIRV_Float16or32 : FloatOfWidths<[16, 32]>;
-def SPIRV_Vector : VectorOfLengthAndType<[2, 3, 4, 8, 16],
+// Remove the vector size restriction
+// Although the vector size can be upto (2^64-1), uint64
+// In tablegen, int is signed int, hence using the upper
+// limit of int64 rather than uint64, it should serve the purpose
+// for all practical cases
+def SPIRV_Vector : VectorOfLengthRangeAndType<[2, 9223372036854775807],
```

Comment:
since spirv is close to hardware, it's make sense to limit the length to certain number and certain upper limit to take advantage of hardware acceleration.
what if we use unlimited length in an upper level dialect and decompose it to limited length in spirv dialect, will this meet our need?